pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -115,3 +194,38 @@ func getInfraMachineFromProvider(platform configv1.PlatformType) (client.Object,
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// shouldMirrorCAPIMachine takes a CAPI machine and determines if there should
+// be a MAPI mirror, it returns true only if:
+//
+// 1. The CAPI Machine is owned by a CAPI MachineSet,
+// 2. That CAPI MachineSet has a MAPI MachineSet Mirror.
+func (r *MachineSyncReconciler) shouldMirrorCAPIMachine(ctx context.Context, machine *capiv1beta1.Machine, logger logr.Logger) (bool, error) {
```

Comment:
I'd make it even more descriptive
```suggestion
func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context.Context, machine *capiv1beta1.Machine, logger logr.Logger) (bool, error) {
```